langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/InternalAzureOpenAiHelper.java

```diff
@@ -119,19 +119,32 @@ public static List<com.azure.ai.openai.models.ChatRequestMessage> toOpenAiMessag
     }
 
     public static com.azure.ai.openai.models.ChatRequestMessage toOpenAiMessage(ChatMessage message) {
-        if (message instanceof AiMessage) {
-            ChatRequestAssistantMessage chatRequestAssistantMessage = new ChatRequestAssistantMessage(getOrDefault(message.text(), ""));
+        if (message.type().messageClass().equals(AiMessage.class)) {
```

Comment:
This can be simplified to `message instanceOf AiMessage`, the same for the rest of types below

Reply:
From a coding perspective, I liked my approach more, but then indeed 1) it's shorter and clearer and 2) I just learned it was more efficient to use instances !